### PR TITLE
feat: adding EdgePreditor and refactoring extractors

### PR DESCRIPTION
**Pull request by andy-ldai**
### EdgePredictor feature
- add EdgePredictor class to call the WebAPI endpoint created by LandingEdge
- create _Extractor class to convert the raw inference JSON to Predictions
- add pytest fixtures to share some validation code

### CODE REVIEW

File: .nojekyll
Comment (AsiaCao) [inline .nojekyll:1]: what is this file used for?

File: landingai/predict.py [hunk 1]
```diff
@@ -193,228 +163,497 @@ def predict(
...
+    session.headers.update(headers)
+    return session
+
+
+def _do_inference(
+    session: Session,
+    endpoint_url: str,
+    files: List[Any],
+    payload: Dict[str, Any],
+    extractorClass: Type[_Extractor],
```
Comment (AsiaCao) [inline landingai/predict.py:646]: nit: should use snake case naming, i.e. `extractor_class`

File: landingai/predict.py [hunk 2]
```diff
@@ -193,228 +163,497 @@ def predict(
...
+            "type": "ClassificationPrediction",
+            "predictions":
+            {
+                "score": 0.9951885938644409,
+                "labelIndex": 0,
+                "labelName": "Fire"
+            }
+        }
+        """
+        print("edge extractor")
```
Comment (AsiaCao) [inline landingai/predict.py:463]: forget to remove?

**Review by AsiaCao — APPROVED** [review]
nice work!

File: .nojekyll
Comment (andy-ldai) [reply .nojekyll:1]: this will skip jekyll github page build, seems we don't need it anymore

Comment (andy-ldai) [reply .nojekyll:1]: removing the file

File: landingai/predict.py (hunk 1 above)
Comment (andy-ldai) [reply landingai/predict.py]: changed

File: landingai/predict.py (hunk 2 above)
Comment (andy-ldai) [reply landingai/predict.py]: nice catch! removed

File: .nojekyll
Comment (andy-ldai) [reply .nojekyll:1]: not used anymore, removing

**Pull request merged**